ledger/pulsemanager/pulsemanager.go

```diff
@@ -147,29 +147,38 @@ func (m *PulseManager) processRecentObjects(
 	drop *jetdrop.JetDrop,
 	dropSerialized []byte,
 ) error {
+	logger := inslogger.FromContext(ctx)
 	m.Recent.ClearZeroTTLObjects()
 	recentObjectsIds := m.Recent.GetObjects()
 	pendingRequestsIds := m.Recent.GetRequests()
-	m.Recent.ClearObjects()
+	defer m.Recent.ClearObjects()
 
 	recentObjects := map[core.RecordID]*message.HotIndex{}
 	pendingRequests := map[core.RecordID][]byte{}
 
 	for id, ttl := range recentObjectsIds {
 		lifeline, err := m.db.GetObjectIndex(ctx, &id, false)
 		if err != nil {
-			inslogger.FromContext(ctx).Error(err)
+			logger.Error(err)
 			continue
 		}
 		encoded, err := index.EncodeObjectLifeline(lifeline)
 		if err != nil {
-			inslogger.FromContext(ctx).Error(err)
+			logger.Error(err)
 			continue
 		}
 		recentObjects[id] = &message.HotIndex{
 			TTL:   ttl,
 			Index: encoded,
 		}
+
+		if m.Recent.IsMine(id) {
+			err := m.db.RemoveObjectIndex(ctx, &id)
+			if err != nil {
```

Comment:
If we have error here, it means something wrong with DB and we have inconsistent state. It's safer to end execution. Furthermore the client has no way of knowing that something went wrong.